geopandas/geodataframe.py

```diff
@@ -1459,7 +1467,15 @@ def _constructor(self):
 
     @property
     def _constructor_sliced(self):
-        return _geoseries_constructor_with_fallback
+        def _geodataframe_constructor_sliced(data=None, index=None, crs=None, **kwargs):
+
```

Comment:
Maybe one final comment: can you add a docstring here documenting a bit the logic we ended up with?

Copied partly from `geoseries.py`, something like:
```
    """
    A specialized (Geo)Series constructor which can fall back to a Series if a certain operation does not produce
    geometries.

    - We only return a GeoSeries if the data is actually of geometry dtype (and so we don't try to convert geometry objects such as the normal GeoSeries(..) constructor does with `_ensure_geometry`). 
    - When we get here from obtaining a row or column from a GeoDataFrame, the goal is to only return a GeoSeries for a geometry column, and not return a GeoSeries if the row happened to come from a DataFrame with only geometry dtype columns (and thus could have a geometry dtype). Therefore, we don't return a GeoSeries if we are sure we are in a row selection case (by checking the identity of the index)
    """
```

Reply:
Added this comment